### ISSUE TITLE
rospack fails to link to boost

### ISSUE THREAD

**Issue by seedship** [ros-melodic-arch/ros-melodic-rospack#3]
When I do yay -S ros-melodic-desktop-full, I get:

-- Build files have been written to: /home/seedship/.cache/yay/ros-melodic-rospack/src/build
Scanning dependencies of target rospack
[ 11%] Building CXX object CMakeFiles/rospack.dir/src/rospack.cpp.o
[ 22%] Building CXX object CMakeFiles/rospack.dir/src/rospack_backcompat.cpp.o
[ 33%] Building CXX object CMakeFiles/rospack.dir/src/rospack_cmdline.cpp.o
[ 44%] Building CXX object CMakeFiles/rospack.dir/src/utils.cpp.o
[ 55%] Linking CXX shared library devel/lib/librospack.so
[ 55%] Built target rospack
Scanning dependencies of target rosstackexe
[ 66%] Building CXX object CMakeFiles/rosstackexe.dir/src/rosstack_main.cpp.o
[ 77%] Linking CXX executable devel/bin/rosstack
/usr/bin/ld: devel/lib/librospack.so: undefined reference to `boost::system::system_category()' /usr/bin/ld: devel/lib/librospack.so: undefined reference to `boost::system::generic_category()'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/rosstackexe.dir/build.make:90: devel/bin/rosstack] Error 1
make[1]: *** [CMakeFiles/Makefile2:249: CMakeFiles/rosstackexe.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
==> ERROR: A failure occurred in build().
Aborting...
[sudo] password for seedship: 
error: target not found: ros-melodic-rospack
error: target not found: ros-melodic-roslib
error: target not found: ros-melodic-rosunit
error: target not found: ros-melodic-roslang
error: target not found: ros-melodic-roslint
error: target not found: ros-melodic-urdf-parser-plugin

**Comment by seedship** [comment]
I just noticed in my install log that I have: boost not satisfied, flushing install queue
What does that mean?

**Comment by seedship** [comment]
I went inside the yay cache, and printed the found ${Boost_LIBRARIES}. It printed Boost::system, but the build is complaining about boost::system.

Is this an issue?

**Comment by acxz** [comment]
Have you reinstalled all of ROS on Arch, yet? Linking issues usually get fixed by reinstalling ROS. And since I am able to install `rospack`, I have a feeling a reinstall will fix this. Just delete all ros packages from your system and go ahead and try reinstalling this package again.

**Comment by jwhendy** [comment]
This will likely break for python3.8 reasons as well. I'm honestly stuck at the moment. When I try to use ROS, I get errors like this:
```
Traceback (most recent call last):
  File "/opt/ros/melodic/lib/python3.8/site-packages/roslaunch/__init__.py", line 322, in main
    p.start()
  File "/opt/ros/melodic/lib/python3.8/site-packages/roslaunch/parent.py", line 273, in start
    self.logger.info("starting roslaunch parent run")
  File "/usr/lib/python3.8/logging/__init__.py", line 1434, in info
    self._log(INFO, msg, args, **kwargs)
  File "/usr/lib/python3.8/logging/__init__.py", line 1565, in _log
    fn, lno, func, sinfo = self.findCaller(stack_info, stacklevel)
TypeError: findCaller() takes from 1 to 2 positional arguments but 3 were given
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1565, in _log
    fn, lno, func, sinfo = self.findCaller(stack_info, stacklevel)
TypeError: findCaller() takes from 1 to 2 positional arguments but 3 were given
```
Trying to rebuild everything pulls from AUR which has stale packages that we've updated here. But building locally is an immense pain as you have to do your own dependency management (aka try to build, look for the failed deps, then go build those one by one). Lastly, I'm unaware of what I even need to rebuild since failing due to PKGBUILD changes is one thing but detecting "all libraries that used to link to something that's now different" seems non-trivial.

**tl;dr**: this is mostly unrelated, but trying to rebuild all of ROS won't work because the AUR doesn't have our changes. @bionade24 we need python3.8 fixes synced to AUR so we can at least handle those and get to the actual issues at hand (if there are any).

**Comment by seedship** [comment]
> Have you reinstalled all of ROS on Arch, yet? Linking issues usually get fixed by reinstalling ROS. And since I am able to install `rospack`, I have a feeling a reinstall will fix this. Just delete all ros packages from your system and go ahead and try reinstalling this package again.

I am trying to fresh install ROS, I never had ROS installed on my system before

**Comment by bionade24** [comment]
@seedship If you around 8 hours, you can simply add my repo and download clean binary packages. 
If you build locally and get errors, clone the AUR Repo and execute `extra-x86_64-build` in it.

**Comment by acxz** [comment]
@bionade24 can you post details on your repo somewhere everyone can see? Maybe like in the pinned AUR comment on the ros-melodic-desktop-full page.

**Comment by acxz** [comment]
@seedship did you use arch4edu by any chance? If so can you try without using it? Maybe their build scripts havent updated yet.

**Comment by bionade24** [comment]
@acxz  I did post it on ROS discourse. I'll post elsewhere when at least ros-melodic-desktop is avaiable again.

**Comment by seedship** [comment]
Ah I found the problem. I had boost 1.65 headers in /usr/local/include as a result from a previous project I was working on. Those headers call functions that were deprecated in boost 1.71, which were my system boost libraries. Thus, the mismatch between header and library caused build problems.

**Comment by acxz** [comment]
Sweet nice to hear!!! Happy ROSing! Please let us know of any issues you encounter.
